storage/src/main/java/tech/pegasys/teku/storage/server/leveldb/LevelDbInstanceFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.leveldb;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.IOException;
+import java.util.Collection;
+import org.fusesource.leveldbjni.JniDBFactory;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.MetricCategory;
+import org.iq80.leveldb.DB;
+import org.iq80.leveldb.Options;
+import tech.pegasys.teku.storage.server.DatabaseStorageException;
+import tech.pegasys.teku.storage.server.rocksdb.RocksDbConfiguration;
+import tech.pegasys.teku.storage.server.rocksdb.core.RocksDbAccessor;
+import tech.pegasys.teku.storage.server.rocksdb.schema.RocksDbColumn;
+
+public class LevelDbInstanceFactory {
+  public static RocksDbAccessor create(
+      final MetricsSystem metricsSystem,
+      final MetricCategory metricCategory,
+      final RocksDbConfiguration configuration,
+      final Collection<RocksDbColumn<?, ?>> columns)
+      throws DatabaseStorageException {
+    checkArgument(
+        columns.stream().map(RocksDbColumn::getId).distinct().count() == columns.size(),
+        "Column IDs are not distinct");
+    final Options options =
+        new Options().createIfMissing(true).maxOpenFiles(configuration.getMaxOpenFiles());
```

Comment:
Just FYI what other options were set in EthereumJ: 
```java
            options.createIfMissing(true);
            options.compressionType(CompressionType.NONE);
            options.blockSize(10 * 1024 * 1024);
            options.writeBufferSize(10 * 1024 * 1024);
            options.cacheSize(0);
            options.paranoidChecks(true);
            options.verifyChecksums(true);
            options.maxOpenFiles(settings.getMaxOpenFiles());
```
I don't think there was some fine tuning since this code was not touched from the very initial version (~7 years old)

Reply:
There was also some code to 'repair' the database. As I remember we had some cases when DB was corrupted and manually calling `repair()` helped. But again this is pretty old code (~5 y.o) and I hope this is not the case in the latest leveldb version

https://github.com/ethereum/ethereumj/blob/11d642750f8127170bc7d6665fe0e6f7f85abec3/ethereumj-core/src/main/java/org/ethereum/datasource/leveldb/LevelDbDataSource.java#L102-L132

Reply:
Also worth to mention that in EthereumJ we managed read/write db cache and did background db flush on the Java side, so the db options above might be suboptimal for our case

Reply:
Yeah I've deliberately stuck with the defaults to see how they go and they seem pretty good so far.  I did experiment with explicitly compacting ranges at one point which is pretty expensive but may be worth it if we can do it reasonably intelligently.

Otherwise I suspect we need to run some perf tests and really understand the trade offs before we start messing with options.